File: 240822/연속되는 수 3/continuous-number3.cpp
#include <iostream>
using namespace std;

int main() {
    int n,max=0;
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++)
        cin>>arr[i];
    for(int i=0,cnt=0;i<n;i++)
    {
        if(i == 0 || arr[i] * arr[i-1] < 0) cnt=0;
        cnt++;
        if(cnt > max) max = cnt;
    }
    cout<<max;
    return 0;
}

File: 240907/연속되는 수 3/continuous-number3.py
n = int(input())

arr = [int(input()) for _ in range(n)]

ans, cnt = 0, 1

for i in range(n):
    if i > 0 and arr[i] * arr[i-1] > 0:
        cnt += 1
    else :
        cnt = 1
    ans = max(ans, cnt)
print(ans)